osm2streets/src/lanes/classic.rs

```diff
@@ -447,11 +473,37 @@ fn add_sidewalks_and_shoulders(
     // For now, model that by putting shoulders.
     if cfg.inferred_sidewalks || tags.is(osm::HIGHWAY, "living_street") {
         if need_fwd_shoulder {
-            fwd_side.push(fwd(&tags, LaneType::Shoulder));
+            fwd_side.push(fwd(highway_type, LaneType::Shoulder));
         }
         if need_back_shoulder {
-            back_side.push(back(&tags, LaneType::Shoulder));
+            back_side.push(back(highway_type, LaneType::Shoulder));
+        }
+    }
+}
+
+fn apply_turn_restrictions(list: &mut Vec<LaneSpec>, value: &str) {
+    // Turn lanes only apply to certain lane types
+    fn applicable(spec: &LaneSpec) -> bool {
+        spec.lt == LaneType::Driving || spec.lt == LaneType::Bus
+    }
+
+    let parts: Vec<&str> = value.split('|').collect();
+    // TODO Warn when things don't match up
+    if parts.len() == list.iter().filter(|l| applicable(*l)).count() {
+        // The parts are ordered from inside to out or left-to-right. The caller always passes them
+        // in the matching order already.
+        let mut parts = parts.into_iter();
+        for spec in list {
+            if applicable(spec) {
+                spec.turn_restrictions = parts
```

Comment:
No attempt yet to interpret these. When we start exploding movements into lane-to-lane operations, that'll be the time. We'd probably want to parse these strings and turn into enums at the very least